File: web/src/javascript/gameState.js
export class Position {
  constructor(obj) {
    this.x = obj.x;
    this.y = obj.y;
  }
}
export class Dimensions {
  constructor(obj) {
    this.width = obj.width;
    this.height = obj.height;
  }
}
export class Cell {
  constructor(obj) {
    this.id = obj.id;
    this.mass = obj.mass;
    this.position = obj.position;
    this.target = obj.target;
  }
}
export class Player {
  constructor(obj) {
    this.id = obj.id;
    this.name = obj.name;
    this.totalMass = obj.totalMass;
    this.cells = obj.cells;
  }
}
export class Food {
  constructor(obj) {
    this.regular = obj.regular;
    this.silver = obj.silver;
    this.gold = obj.gold;
  }
}
export class GameState {
  constructor(obj) {
    this.id = obj.id;
    this.tick = obj.tick;
    this.players = obj.players;
    this.food = obj.food;
    this.map = obj.map;
    this.viruses = obj.viruses;
  }
}

function createPlayers()

export default function parseServerState(response) {
  const data = response.data;
  
}

File: game/src/main/scala/io/aigar/game/serializable/GameState.scala
package io.aigar.game.serializable

import com.github.jpbetz.subspace.Vector2

/**
 * Serializable classes that represent the current state of a game. These are
 * the classes that will be sent over the network, so they should follow the
 * conventions of the API.
 */

case class Position(
  x: Float,
  y: Float
)
case class Dimensions(
  width: Int,
  height: Int
)
case class Cell(
  id: Int,
  mass: Int,
  radius: Int,
  position: Position,
  target: Position,
  burst: Boolean
)
case class Player(
  id: Int,
  name: String,
  total_mass: Integer,
  isActive: Boolean,
  cells: List[Cell]
)
case class Resources(
  regular: List[Vector2],
  silver: List[Vector2],
  gold: List[Vector2]
)
case class Virus(
  position: Position,
  mass: Int,
  radius: Int
)
case class GameState(
  id: Int,
  var paused: Boolean,
  var disabledLeaderboard: Boolean,
  multiplier: Int,
  tick: Int,
  timeLeft: Float,
  players: List[Player],
  resources: Resources,
  map: Dimensions,
  viruses: List[Virus]
)
